cli/explain/compactgraph/compactgraph.go

```diff
@@ -354,9 +356,16 @@ func flattenInvalidates(invalidates []any, isTool bool) map[string]uint32 {
 				}
 				transitivelyInvalidated[n.Mnemonic+suffix]++
 			}
+		case []any:
+			// invalidates and any slices it transitively references are reachable and not modified while traversing, so
+			// we can identify them with the pointer to their slice data.
+			ptr := unsafe.SliceData(n)
+			if _, seen := slicesSeen[ptr]; !seen {
+				slicesSeen[ptr] = struct{}{}
+				toVisit = append(toVisit, n...)
+			}
 		default:
-			// If n is not a Spawn, it must be a slice of Spawns or slices.
-			toVisit = append(toVisit, n.([]any)...)
+			log.Fatalf("unexpected type in invalidates: %T", n)
```

Comment:
i wonder if we can use a custom type/alias for slice of spawns and slices of slice here. Then we and use a pointer to that type instead of having to use unsafe pointers.

Reply:
I tried this but it doesn't end up fixing the issue since the newtype wrappers around the same slice will all be distinct instances and thus the "seen" check becomes pointless.